### DISCUSSION TITLE
Rate Too High > Max Rate berechnen

### DISCUSSION THREAD

**Discussion by Schnubl** [jomjol/AI-on-the-edge-device#2983] [Q&A] [answered]
Hallo Zusammen,
ich möchte euch um Hilfestellung bitten.
Ich möchte gerene einen Gaszähler auslesen, Grundsätzlich funktioniert das auch aber ich bekomme dauernd Rate to High.

Der Standardwert im Feld Maximum Rate Value ist 0,05 und das ist  vermutlich viel zu neidrig bei einem Gaszähler.
Kann mir jemand erklären wie das berechnet wirdd, oder wie ich einen passenden Max Wert bei mir setzen kann damit ich den Rate to High Felder nicht mehr bekommme aber trotzdem noch eine Prüffrung auf komplett falsche Werte habe?

Git es ein How To bei dem ich mir diesen Wert für mich speziell berechnnen kann?

Danke euch!

**Comment by SybexX** [comment]
https://jomjol.github.io/AI-on-the-edge-device-docs/Correction%20Algorithm/#maxratevalue-and-maxratetype

**Reply by Schnubl** [reply]
Danke SybexX,
wenn auf dem Zähler steht Qmax (maximaler Durchfluss) 6m3/h würde das dann ja bedeuten das er max 0,5m3 in 5 Minuten zusammmenbringen kann. 
Müsste dann der Wert für Max Rate Value  nicht ca. 0,5 heissen?

Mir wurden aber schon >5 angezeigt bei Rate too High

Oder bin ich da total auf dem Holzweg?

**Comment by SybexX** [comment]
was hast du bei Maximum Rate Type eingestellt, RateChange(pro minute) oder AbsoluteChange(pro Runde)?

**Reply by Schnubl** [reply]
Das ist aktuell nicht angehakt und bei Max Rate Value hab ich jetzt einfach mal 5 reingeschieben damit er nicht immer abbricht. Aber das ist auf dauer vermutlich nicht ideal.
Soll ich den Rate Type anhaken, wenn ja welchen Typ soll ich auswählen?

Danke für die Hilfe.

**Comment by SybexX** [comment]
ich glaube wenn du Maximum Rate Type nicht aktivierst, wird AbsoluteChange(pro Runde) verwendet.
Lässt du dir den Wert in m3 oder in Liter ausgeben? den 0,5m3 sind ja 500 Liter

**Comment by SybexX** [comment]
1.	am besten bei voller Belastung (Heizung läuft,....) die Werterhöhung pro Minute messen.
2.	jenachdem ob man bei Maximum Rate Type, RateChange oder AbsoluteChange eingestellt hat:
	- RateChange ==>	Maximum Rate Value = Erfasster Wert
	- AbsoluteChange ==>	Maximum Rate Value = Erfasster Wert * Rundenzeit

**Reply by Schnubl** [reply]
AbsoluteChange wäre dann die zeit die das hinterste Rad für eine Umdrehung benötigt * die Erfasste Differenz in dieser Zeit?
RateChange wäre die zeit die in 5 Minuten durchgelaufen ist?

hab ich das jetzt richtig verstanden?

**Comment by SybexX** [comment, marked as answer]
AbsoluteChange = pro Runde/Lesung, bei dir offenbar alle 5 Minuten ==>> Verbrauch in 5 Minuten ==>> 0,5m3 oder 500 Liter
RateChange = pro Minute ===>> Verbrauch pro Minute ==>> 0,1m3 oder 100 Liter

die Berechnung sieht in etwa so asu:

Verbrauch pro Minute
-----------------------
Maximum Rate Type = RateChange
	Differenz bzw. Vergleichswert = (gelesener Wert - Vorwert) / verstrichene Minuten

Verbrauch pro Runde/Lesung/Auswertung
-------------------------------------------
Maximum Rate Type = AbsoluteChange		
	Differenz bzw. Vergleichswert = gelesener Wert - Vorwert

wenn Differenz bzw. Vergleichswert grösser Maximum Rate Value, dann gib ein Fehler aus.

**Reply by Schnubl** [reply]
Super danke, ich werd das morgen mal messen und dann schaeun ob das mit dem rechnerischen Wert in irgend einer Weise zusammenpasst.

Vielen Dank für deine Hilfe.

**Comment by stonehage** [comment]
Hier nochmal ein Hinweis, hab ein bischen tüffteln müssen bevor ich gecheckt habe das eh alles aufgedruckt ist ;-)

![gasmeter](https://github.com/user-attachments/assets/c87e2fd6-16bc-4d2d-a2d5-21b3723fe150)

Habe bei mir dann den Wert 0,5 für 5min eingestellt.

6 m³ /h = 0,1 m³ /min  => 0,5m³ für 5min

Damit ergeben sich zwei Lösungen:

RateChange -> 0,1
AbsoluteChange -> 0,5
